<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="FloWuenne" />

<meta name="date" content="2023-06-12" />

<title>Filter proteomics data</title>

<script src="site_libs/header-attrs-2.21/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/united.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">mi_spatialomics</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="data_processing.html">Data processing</a>
</li>
<li>
  <a href="data_analysis.html">Data analysis</a>
</li>
<li>
  <a href="figures.html">Figures</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Filter proteomics data</h1>
<h4 class="author">FloWuenne</h4>
<h4 class="date">2023-06-12</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2023-06-14
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>mi_spatialomics/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.0). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20230612code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20230612)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20230612code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20230612)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstree2d015dd6457db41a75484766389597ec8f5f45e3targetblank2d015dda">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/2d015dd6457db41a75484766389597ec8f5f45e3" target="_blank">2d015dd</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomFloWuennemispatialomicstree2d015dd6457db41a75484766389597ec8f5f45e3targetblank2d015dda"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/FloWuenne/mi_spatialomics/tree/2d015dd6457db41a75484766389597ec8f5f45e3" target="_blank">2d015dd</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    data/.DS_Store
    Ignored:    references/
    Ignored:    renv/library/
    Ignored:    renv/staging/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/proteomics.filter_proteomic_data.Rmd</code>) and HTML
(<code>docs/proteomics.filter_proteomic_data.html</code>) files. If
you’ve configured a remote Git repository (see
<code>?wflow_git_remote</code>), click on the hyperlinks in the table
below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/2d015dd6457db41a75484766389597ec8f5f45e3/analysis/proteomics.filter_proteomic_data.Rmd" target="_blank">2d015dd</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-06-14
</td>
<td>
Finalized imputation with RF.
</td>
</tr>
<tr>
<td>
html
</td>
<td>
<a href="https://rawcdn.githack.com/FloWuenne/mi_spatialomics/2d015dd6457db41a75484766389597ec8f5f45e3/docs/proteomics.filter_proteomic_data.html" target="_blank">2d015dd</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-06-14
</td>
<td>
Finalized imputation with RF.
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="introduction" class="section level1">
<h1>Introduction</h1>
<p>Here we will process and analyze the data generated via
laser-microdissection coupled with high-sensitiviy proteomics of the
endocardial layer from control hearts and hearts one day after
myocardial infarction. We have a total of 3 groups in this dataset, all
representing laser microdissected regions of endocardium from mouse
hearts: 1) control = Mouse hearts without infarct 2) MI_IZ = Mouse
hearts 1 day post MI. Endocardial region adjacent to the infarct. 3)
MI_remote = Mouse hearts 1 day post MI. Endocardial region remote to the
infarct.</p>
</div>
<div id="load-data" class="section level1">
<h1>Load data</h1>
<p>First, let’s load the raw data and the associated metadata for all
samples.</p>
<pre class="r"><code>prot_res &lt;- fread(&quot;./data/proteomics_endocardial_layer.tsv&quot;)
metadata &lt;- fread(&quot;./data/metadata.proteomics_endocardial_layer.txt&quot;)</code></pre>
</div>
<div id="analysis" class="section level1">
<h1>Analysis</h1>
<div id="calculate-missingness-per-sample-na-values"
class="section level2">
<h2>Calculate missingness per sample (# NA values)</h2>
<p>As a first QC, we will calculate the amount of missing data (NA
values) per sample in the proteomics data.</p>
<pre class="r"><code>missing &lt;- colSums(is.na(prot_res))[6:ncol(prot_res)]
non_missing &lt;-colSums(!is.na(prot_res))[6:ncol(prot_res)]
missingness &lt;- missing / (non_missing + missing)
missingness &lt;- as.data.frame(missingness)
missingness$sample &lt;- rownames(missingness)

## Add group to counts
missingness &lt;- missingness %&gt;%
  mutate(&quot;group&quot; = if_else(grepl(&quot;control&quot;,sample),&quot;control&quot;,
                           if_else(grepl(&quot;MI_IZ&quot;,sample),&quot;MI_IZ&quot;,&quot;MI_remote&quot;)))
## Set order of groups
missingness$group &lt;- factor(missingness$group,
                                levels = c(&quot;control&quot;,&quot;MI_remote&quot;,&quot;MI_IZ&quot;))

avg_missingness &lt;- mean(missingness$missingness)

ggplot(missingness,aes(sample,missingness,fill = group)) +
  geom_bar(stat = &quot;identity&quot;) +
  coord_flip() +
  scale_fill_npg() +
  labs(x = &quot;Samples&quot;,
       y = &quot;% missing values&quot;) +
  geom_hline(yintercept = avg_missingness, linetype = 2)</code></pre>
<p><img src="figure/proteomics.filter_proteomic_data.Rmd/unnamed-chunk-2-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-2-1">
Past versions of unnamed-chunk-2-1.png
</button>
</p>
<div id="fig-unnamed-chunk-2-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/2d015dd6457db41a75484766389597ec8f5f45e3/docs/figure/proteomics.filter_proteomic_data.Rmd/unnamed-chunk-2-1.png" target="_blank">2d015dd</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-06-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="filter-out-proteins-that-are-identified-in-3-mi-samples"
class="section level2">
<h2>Filter out proteins that are identified in &lt; 3 MI samples</h2>
<p>Then we are going to calculate the missingness per sample group (the
treatment group and area that was excised by laser microdissection) and
exclude proteins that are only identified in a subset of samples per
group. For controls, we expect a protein to be measured in at least 2/3
samples (33% missingness allowed). For MI samples, we expect a protein
to be measured in at least 2/4 samples (50% missingness allowed)</p>
<pre class="r"><code>na_mi &lt;- prot_res %&gt;%
  pivot_longer(control_r1:MI_remote_r4,
               names_to = &quot;sample&quot;,
               values_to = &quot;protein_exp&quot;) %&gt;%
  mutate(&quot;group&quot; = if_else(grepl(&quot;control&quot;,sample),&quot;control&quot;,
                           if_else(grepl(&quot;MI_IZ&quot;,sample),&quot;MI_IZ&quot;,&quot;MI_remote&quot;))) %&gt;%
  group_by(Protein_Ids,Genes,group) %&gt;%
  summarise(na_count = sum(is.na(protein_exp))) %&gt;%
  ungroup() %&gt;%
  mutate(&quot;percent_na&quot; = if_else(group == &quot;control&quot;,na_count / 3, na_count / 4)) %&gt;%
  select(-na_count) %&gt;% ## control: n = 3 and MI: n = 4
  pivot_wider(names_from = &quot;group&quot;,values_from = percent_na) %&gt;%
  mutate(&quot;retain&quot; = if_else(control &lt;= 0.3 | MI_remote &lt;= 0.25 | MI_IZ &lt;= 0.25, &quot;yes&quot;,&quot;no&quot;))</code></pre>
<pre><code>`summarise()` has grouped output by &#39;Protein_Ids&#39;, &#39;Genes&#39;. You can override
using the `.groups` argument.</code></pre>
<pre class="r"><code>prot_res_filt &lt;- subset(prot_res,Protein_Ids %in% subset(na_mi,retain == &quot;yes&quot;)$Protein_Ids)</code></pre>
</div>
<div id="exclude-contaminants-keratins-etc." class="section level2">
<h2>Exclude contaminants (keratins etc.)</h2>
<p>After filtering based on missingness, we will filter out any
proteions that are considered contaminants based on know contaminations
based on <a
href="https://pubs.acs.org/doi/10.1021/acs.jproteome.2c00145">Frankenfeld
2022</a></p>
<pre class="r"><code>## Count number of detected proteins with contaminants
prot_res_quant_cont &lt;- prot_res_filt %&gt;%
  select(control_r1:MI_remote_r4) %&gt;%
  summarise_all(funs(sum(!is.na(.))))</code></pre>
<pre><code>Warning: `funs()` was deprecated in dplyr 0.8.0.
ℹ Please use a list of either functions or lambdas:

# Simple named list: list(mean = mean, median = median)

# Auto named with `tibble::lst()`: tibble::lst(mean, median)

# Using lambdas list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre class="r"><code>## Count number of detected proteins without contaminants
prot_res_filt &lt;- prot_res_filt %&gt;%
  mutate(&quot;contaminant&quot; = if_else(grepl(&quot;Cont&quot;,Protein_Ids),&quot;yes&quot;,&quot;no&quot;))

prot_res_quant &lt;-  prot_res_filt %&gt;%
  subset(contaminant == &quot;no&quot;) %&gt;%
  select(control_r1:MI_remote_r4) %&gt;%
  summarise_all(funs(sum(!is.na(.))))</code></pre>
<pre><code>Warning: `funs()` was deprecated in dplyr 0.8.0.
ℹ Please use a list of either functions or lambdas:

# Simple named list: list(mean = mean, median = median)

# Auto named with `tibble::lst()`: tibble::lst(mean, median)

# Using lambdas list(~ mean(., trim = .2), ~ median(., na.rm = TRUE))
Call `lifecycle::last_lifecycle_warnings()` to see where this warning was
generated.</code></pre>
<pre class="r"><code>final_counts &lt;- t(rbind(prot_res_quant_cont,
                      prot_res_quant))</code></pre>
</div>
<div id="merge-metadata-with-protein-counts-per-sample"
class="section level2">
<h2>Merge metadata with protein counts per sample</h2>
<p>Here, we will plot the number of proteins identified per group.</p>
<pre class="r"><code>colnames(final_counts) &lt;- c(&quot;w_cont&quot;,&quot;wo_cont&quot;)
final_counts &lt;- as.data.frame(final_counts) %&gt;%
  mutate(&quot;sample&quot; = rownames(final_counts))

## Add protein counts to metadata
metadata_counts &lt;- left_join(metadata,final_counts, by = &quot;sample&quot;)

## Add group to counts
metadata_counts &lt;- metadata_counts %&gt;%
  mutate(&quot;group&quot; = if_else(grepl(&quot;control&quot;,sample),&quot;control&quot;,
                           if_else(grepl(&quot;MI_IZ&quot;,sample),&quot;MI_IZ&quot;,&quot;MI_remote&quot;)))

## Set order of groups
metadata_counts$group &lt;- factor(metadata_counts$group,
                                levels = c(&quot;control&quot;,&quot;MI_remote&quot;,&quot;MI_IZ&quot;))

## Plot number of proteins without contaminants per sample
ggbarplot(metadata_counts,
          x = &quot;group&quot;,
          y = &quot;wo_cont&quot;,
          fill = &quot;group&quot;,
          add = c(&quot;mean_se&quot;),
          error.plot = &quot;upper_errorbar&quot;,
          color = &quot;black&quot;,
          palette = &quot;npg&quot;) +
  theme_minimal_hgrid()  +
  labs (x = &quot;Treatment group&quot;,
         y = &quot;# Proteins identified&quot;) +
  font(&quot;xlab&quot;, size = 16, color = &quot;black&quot;, face = &quot;bold&quot;) +
  font(&quot;ylab&quot;, size = 16, color = &quot;black&quot;, face = &quot;bold&quot;) +
  scale_y_continuous(
    expand = expansion(mult = c(0, 0.05))
  ) +
  rremove(&quot;legend&quot;)</code></pre>
<p><img src="figure/proteomics.filter_proteomic_data.Rmd/unnamed-chunk-5-1.png" width="672" style="display: block; margin: auto;" /></p>
<p>
<button type="button" class="btn btn-default btn-xs btn-workflowr btn-workflowr-fig" data-toggle="collapse" data-target="#fig-unnamed-chunk-5-1">
Past versions of unnamed-chunk-5-1.png
</button>
</p>
<div id="fig-unnamed-chunk-5-1" class="collapse">
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<a href="https://github.com/FloWuenne/mi_spatialomics/blob/2d015dd6457db41a75484766389597ec8f5f45e3/docs/figure/proteomics.filter_proteomic_data.Rmd/unnamed-chunk-5-1.png" target="_blank">2d015dd</a>
</td>
<td>
FloWuenne
</td>
<td>
2023-06-14
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div id="create-final-filtered-protein-table" class="section level2">
<h2>Create final filtered protein table</h2>
<p>Finally, we will create a final filtered protein table, that we will
use for imputation.</p>
<pre class="r"><code>prot_res_final &lt;- subset(prot_res_filt,contaminant == &quot;no&quot;) %&gt;%
  select(-c(contaminant,First_Protein_Description))

write.table(prot_res_final,
            file = &quot;./output/proteomics.filtered_proteins.tsv&quot;,
            sep = &quot;\t&quot;,
            col.names = TRUE,
            row.names = FALSE,
            quote = FALSE)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.2.3 (2023-03-15)
Platform: aarch64-apple-darwin20 (64-bit)
Running under: macOS Ventura 13.4

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.2-arm64/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices datasets  utils     methods   base     

other attached packages:
 [1] ggsci_3.0.0       here_1.0.1        cowplot_1.1.1     ggpubr_0.6.0     
 [5] data.table_1.14.8 lubridate_1.9.2   forcats_1.0.0     stringr_1.5.0    
 [9] dplyr_1.1.2       purrr_1.0.1       readr_2.1.4       tidyr_1.3.0      
[13] tibble_3.2.1      ggplot2_3.4.2     tidyverse_2.0.0   workflowr_1.7.0  

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.10      getPass_0.2-2    ps_1.7.4         rprojroot_2.0.3 
 [5] digest_0.6.31    utf8_1.2.3       R6_2.5.1         backports_1.4.1 
 [9] evaluate_0.21    highr_0.10       httr_1.4.6       pillar_1.9.0    
[13] rlang_1.1.1      rstudioapi_0.14  whisker_0.4.1    car_3.1-2       
[17] callr_3.7.3      jquerylib_0.1.4  rmarkdown_2.21   labeling_0.4.2  
[21] munsell_0.5.0    broom_1.0.5      compiler_4.2.3   httpuv_1.6.11   
[25] xfun_0.39        pkgconfig_2.0.3  htmltools_0.5.5  tidyselect_1.2.0
[29] fansi_1.0.4      crayon_1.5.2     tzdb_0.4.0       withr_2.5.0     
[33] later_1.3.1      grid_4.2.3       jsonlite_1.8.4   gtable_0.3.3    
[37] lifecycle_1.0.3  git2r_0.32.0     magrittr_2.0.3   scales_1.2.1    
[41] cli_3.6.1        stringi_1.7.12   cachem_1.0.8     carData_3.0-5   
[45] farver_2.1.1     renv_0.17.3      ggsignif_0.6.4   fs_1.6.2        
[49] promises_1.2.0.1 bslib_0.4.2      generics_0.1.3   vctrs_0.6.2     
[53] tools_4.2.3      glue_1.6.2       hms_1.1.3        processx_3.8.0  
[57] abind_1.4-5      fastmap_1.1.1    yaml_2.3.7       timechange_0.2.0
[61] colorspace_2.1-0 rstatix_0.7.2    knitr_1.42       sass_0.4.6      </code></pre>
</div>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
